pkgs/development/libraries/gbenchmark/default.nix

```diff
@@ -2,31 +2,40 @@
 , stdenv
 , fetchFromGitHub
 , cmake
+, ninja
 , gtest
 , prometheus-cpp
 }:
 
 stdenv.mkDerivation rec {
   pname = "gbenchmark";
-  version = "1.8.5";
+  version = "1.9.0";
 
   src = fetchFromGitHub {
     owner = "google";
     repo = "benchmark";
     rev = "v${version}";
-    hash = "sha256-c46Xna/t21WKaFa7n4ieIacsrxJ+15uGNYWCUVuUhsI=";
+    hash = "sha256-5cl1PIjhXaL58kSyWZXRWLq6BITS2BwEovPhwvk2e18=";
   };
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake ninja ];
+
+  cmakeFlags = [
+    # We ran into issues with gtest 1.8.5 conditioning on
+    # `#if __has_cpp_attribute(maybe_unused)`, which was, for some
+    # reason, going through even when C++14 was being used and
+    # breaking the build on Darwin by triggering warnings about using
+    # C++17 features.
+    #
+    # This might be a problem with our Clang, as it does not reproduce
+    # with Xcode, but since `-Werror` is painful for us anyway and
+    # upstream exposes a CMake flag to turn it off, we just use that.
+    (lib.cmakeBool "BENCHMARK_ENABLE_WERROR" false)
```

Comment:
Honestly I'm not sure this needs the comment — disabling -Werror would be a sensible thing to do notwithstanding this issue, just because it's -Werror.

Reply:
That’s fair. I can drop it if you would prefer, but I like build system tweaks to be motivated for future spelunkers and I’m nervous about our compiler acting weird and causing more problems down the line. Perhaps it’d be better in the commit message?

Reply:
Yeah, I think it'd be better in the commit message — this looks too much like the only reason for Werror is this one attribute problem, when really it should be policy that we disable Werror across the board.

Reply:
Bit late now, sadly :( but if you want me to send a PR dropping the comment I can.